ComponentKit/Core/CKComponent.mm

```diff
@@ -140,9 +141,21 @@ - (CKComponentViewContext)viewContext
       CKAssert(v.ck_component == self, @"");
     }
 
-    const CGPoint anchorPoint = v.layer.anchorPoint;
-    [v setCenter:effectiveContext.position + CGPoint({size.width * anchorPoint.x, size.height * anchorPoint.y})];
-    [v setBounds:{v.bounds.origin, size}];
+    @try {
+      const CGPoint anchorPoint = v.layer.anchorPoint;
+      [v setCenter:effectiveContext.position + CGPoint({size.width * anchorPoint.x, size.height * anchorPoint.y})];
+      [v setBounds:{v.bounds.origin, size}];
+    } @catch (NSException *exception) {
+      NSMutableArray<CKComponent *> *const componentBacktrace = [NSMutableArray arrayWithObject:supercomponent];
+      while ([componentBacktrace lastObject]
+             && [componentBacktrace lastObject]->_mountInfo
+             && [componentBacktrace lastObject]->_mountInfo->supercomponent) {
+        [componentBacktrace addObject:[componentBacktrace lastObject]->_mountInfo->supercomponent];
+      }
```

Comment:
@eczarny agree, leave it as is then. perhaps pull it into a static C function at the bottom of this file, just before @end?